src/Utilities/ModernStringUtils.hpp

```diff
@@ -30,6 +34,36 @@ namespace qmcplusplus
  *  *  For other XML derived text this should never be used since we should assume that to be UTF-8 encoded.
  */
 std::string lowerCase(const std::string_view s);
+
+/** prevent clash with string_utils.h */
+namespace modernstrutil {
+  /** return string_view tokens */
+  std::vector<std::string_view> split(const std::string_view s, const std::string_view delimiters);
+  /** remove white space from each beginning and end of string_view */
+  std::string_view strip(const std::string_view s);
+}
+
+/** alternate to string2real
+ *  calls c++ string to real conversion based on T's precision template<typename T>
+ */
+template<typename T>
+inline T string2Real(const std::string_view svalue) {
+  static_assert(std::is_floating_point_v<T>);
+      T result;
+// full support for floating point from char not present until stdlibc++ aligned with gcc11
+// there is still not from_char for floats as of libc++ 15
+#if _GLIBCXX_RELEASE > 10
+  auto [prt, ec] = std::from_chars(svalue.data(), svalue.data() + svalue.size(), result, std::chars_format::general);
```

Comment:
What is the benefit of this special case? Faster? I don't mind keep only the universal working way although a bit slow.

Reply:
It's c++ 17 and the point of this file is to use the best practice for the current standard requirement of the code.  But here we have to make an except due to incomplete support for the std library.

The new way does not require a mutable string to be constructed and in general fits with the string_view and current c++ io framework the other is a c function.

Reply:
C++17 language yes. C++17 standard library, if it is a hassle due to implementation issue, I prefer using a dumb method. Is your new way safer because of error handling? That could shift my preference.

Reply:
Yes it is able to clearly indicate an error.  atof really can't do anything you get 0.0 or UB.  

So with the c++17 way if it can't do the conversion or its out of range for the floating point type we can throw.  This means we can deal with bad input better.